pkg/minikube/kubeconfig/config.go

```diff
@@ -178,3 +183,66 @@ func decode(data []byte) (*api.Config, error) {
 
 	return config.(*api.Config), nil
 }
+
+// GetKubeConfigStatus verifys the ip stored in kubeconfig.
+func GetKubeConfigStatus(ip net.IP, filename string) (string, error) {
+	if ip == nil {
+		return "", fmt.Errorf("Error, empty ip passed")
+	}
+	kip, err := getIPFromKubeConfig(filename)
+	if err != nil {
+		return "", err
+	}
+	if kip.Equal(ip) {
+		return "Correctly Configured: pointing to minikube-vm at " + kip.String(), nil
+	}
+	return "Misconfigured: pointing to stale minikube-vm at " + kip.String() +
+		"\nTo fix the kubectl context, run minikube update-context", nil
+
+}
+
+// UpdateKubeconfigIP overwrites the IP stored in kubeconfig with the provided IP.
+func UpdateKubeconfigIP(ip net.IP, filename string) (string, error) {
+	if ip == nil {
+		return "", fmt.Errorf("Error, empty ip passed")
+	}
+	kip, err := getIPFromKubeConfig(filename)
+	if err != nil {
+		return "", err
+	}
+	if kip.Equal(ip) {
+		return "Correctly Configured: pointing to minikube-vm at " + kip.String(), nil
+	}
+	con, err := ReadConfigOrNew(filename)
+	if err != nil {
+		return "", errors.Wrap(err, "Error getting kubeconfig status")
+	}
+	con.Clusters["minikube"].Server = "https://" + ip.String() + ":" + strconv.Itoa(constants.APIServerPort)
+	err = WriteConfig(con, filename)
+	if err != nil {
+		return "Unable to reconfigure Kubeconfig IP", nil
+	}
+	return "Reconfigured: pointing to minikube-vm at " + ip.String(), nil
+}
+
+// getIPFromKubeConfig returns the IP address stored for minikube in the kubeconfig specified
+func getIPFromKubeConfig(filename string) (net.IP, error) {
```

Comment:
Yes.  @abbytiz minikube allows for multiple machines to be named.  The name 'minikube' is the default but it can hold any value.  You can test that your code is working with this by launching minikube using 'minikube start --profile test'.  This will spin up a second copy of  minikube with its own name and kubeconfig using that name.